### PR TITLE
Fix Workflow Global Permissions Nil Check

### PR DESCRIPTION
**Pull request by rohankh532**
Fix for ossf/scorecard-action#195

### CODE REVIEW

File: signing/signing.go [hunk 1]
```diff
@@ -232,8 +232,8 @@ func verifyScorecardWorkflow(workflowContent string) error {
 	}
 
 	// Verify that the all scope, if set, isn't write-all.
-	globalPermAll := workflow.Permissions.All
-	if globalPermAll != nil && globalPermAll.Value == "write-all" {
+	globalPerm := workflow.Permissions
```
Comment (azeemshaikh38) [inline signing/signing.go:235]: Might need a similar check on line 230. `workflow == nil`

**Review by azeemshaikh38 — COMMENTED** [review]
Could you add tests for this? Especially the verifyScorecardWorkflow fn.

File: signing/signing.go (hunk 1 above)
Comment (laurentsimon) [reply signing/signing.go:235]: and at line 241 which also accesses the `Permissions` field

Comment (rohankh532) [reply signing/signing.go]: Fixed + added tests

**Review by laurentsimon — APPROVED** [review]

File: signing/signing_test.go [hunk 2]
```diff
@@ -46,10 +46,17 @@ func TestVerifySignatureInvalidRepo(t *testing.T) {
...
-	assert.Equal(t, err, nil)
+func TestVerifyValidWorkflows(t *testing.T) {
+	workflowFiles := []string{
+		"../testdata/workflow-valid.yml",
+		"../testdata/workflow-valid-noglobalperm.yml",
+	}
+
+	for _, workflowFile := range workflowFiles {
+		workflowContent, _ := ioutil.ReadFile(workflowFile)
+		err := verifyScorecardWorkflow(string(workflowContent))
```
Comment (azeemshaikh38) [inline signing/signing_test.go:57]: Also add a test for empty file.

**Review by azeemshaikh38 — APPROVED** [review]

File: signing/signing_test.go (hunk 2 above)
Comment (rohankh532) [reply signing/signing_test.go:57]: Added
